Index: Themes/Storefront/resources/lang/en/layout.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Themes/Storefront/resources/lang/en/layout.php	(date 1560515392000)
+++ Themes/Storefront/resources/lang/en/layout.php	(date 1560520037000)
@@ -10,6 +10,7 @@
     'product_tags' => 'Product Tags',
     'enter_your_email' => 'Enter your email...',
     'subscribe' => 'Subscribe',
+    'price_confirm' => 'Please confirm price recalculation',
     'mini_cart' => [
         'my_cart' => 'My Cart',
         'your_cart_is_empty' => 'Your cart is empty',
Index: Modules/Product/Entities/Product.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Modules/Product/Entities/Product.php	(date 1560515392000)
+++ Modules/Product/Entities/Product.php	(date 1560518711000)
@@ -272,7 +272,20 @@
 
     public function getSellingPriceAttribute($sellingPrice)
     {
-        return Money::inDefaultCurrency($sellingPrice);
+        $currentContract = Session::get('CurrentContract');
+        if ($currentContract) {
+            $contractModel = Contract::find($currentContract);
+            $pricelistId = $contractModel->pricelist_id;
+            $priceModel = Pricing::where(['product_id' => $this->id, 'pricelist_id' => $pricelistId])->first();
+            if ($priceModel) {
+                $newprice = $priceModel->price;
+            } else {
+                $newprice = $sellingPrice;
+            }
+        } else {
+            $newprice = $sellingPrice;
+        }
+        return Money::inDefaultCurrency($newprice);
     }
 
     public function getTotalAttribute($total)
Index: Modules/Contract/Http/Controllers/ContractController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Modules/Contract/Http/Controllers/ContractController.php	(date 1560515392000)
+++ Modules/Contract/Http/Controllers/ContractController.php	(date 1560519825000)
@@ -2,6 +2,7 @@
 
 namespace Modules\Contract\Http\Controllers;
 
+use http\Client\Response;
 use Illuminate\Routing\Controller;
 use Illuminate\Support\Facades\Session;
 use Modules\Contract\Entities\Contract;
@@ -20,9 +21,15 @@
 
     public function contractUpdate($id) {
         $contract = Session::get("CurrentContract");
+        $prevContractModel = Contract::whereId($contract)->first();
+        $currentContractModel = Contract::whereId($id)->first();
         if($contract != $id){
             Session::put("CurrentContract", $id);
-            return "$id Ok";
+            if ($prevContractModel->pricelist_id == $currentContractModel->pricelist_id) {
+                return \response()->json(["None", $contract, trans('storefront::layout.price_confirm')]);
+            } else {
+                return \response()->json(["Ok", $contract, trans('storefront::layout.price_confirm')]);
+            }
         }
     }
 }
Index: Themes/Storefront/views/public/partials/top_nav.blade.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Themes/Storefront/views/public/partials/top_nav.blade.php	(date 1560515392000)
+++ Themes/Storefront/views/public/partials/top_nav.blade.php	(date 1560520213000)
@@ -70,7 +70,17 @@
 <script>
     function sendData(url) {
         $.get(url, function(data){
-            location.reload();
+            if (data[0] == 'Ok'){
+                var result = confirm(data[2]);
+                if (result) {
+                    location.reload();
+                } else {
+                    $.get('contract/' + data[1]);
+                }
+            } else {
+                location.reload();
+            }
+            // location.reload();
         });
     }
 </script>
Index: Modules/Cart/Http/Controllers/CartController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Modules/Cart/Http/Controllers/CartController.php	(date 1560515392000)
+++ Modules/Cart/Http/Controllers/CartController.php	(date 1560518588000)
@@ -18,6 +18,7 @@
     {
         $cart = Cart::instance();
         $countries = Country::supported();
+        $cart->updateContract();
 
         return view('public.cart.index', compact('cart', 'countries'));
     }
